File: TrueVote.Services/Database/Korisnik.cs
using System;
using System.Collections.Generic;

namespace TrueVote.Services.Database;

public partial class Korisnik
{
    public int Id { get; set; }

    public string Ime { get; set; } = null!;

    public string Prezime { get; set; } = null!;

    public string? Email { get; set; }

    public string KorisnickoIme { get; set; } = null!;

    public string? PasswordSalt { get; set; }

    public string? PasswordHash { get; set; }

    public int UlogaId { get; set; }

    public int OpstinaId { get; set; }

    public byte[]? Slika { get; set; }

    public string? Pin { get; set; }

    public bool Obrisan { get; set; }

    public virtual ICollection<Glas> Glas { get; set; } = new List<Glas>();

    public virtual Opstina Opstina { get; set; } = null!;

    public virtual Uloga Uloga { get; set; } = null!;
}

File: UI/truevote_mobile/lib/models/korisnik.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:truevote_mobile/models/opstina.dart';
import 'package:truevote_mobile/models/uloga.dart';

part 'korisnik.g.dart';

@JsonSerializable()
class Korisnik {
  int id;
  String? ime;
  String? prezime;
  String? email;
  String? korisnickoIme;
  int? ulogaId;
  int? opstinaId;
  String? slika;
  String? pin;
  Opstina? opstina;
  Uloga? uloga;

  Korisnik({
    required this.id,
    this.ime,
    this.prezime,
    this.email,
    this.korisnickoIme,
    this.ulogaId,
    this.opstinaId,
    this.slika,
    this.pin,
    this.opstina,
    this.uloga,
  });

  factory Korisnik.fromJson(Map<String, dynamic> json) =>
      _$KorisnikFromJson(json);

  Map<String, dynamic> toJson() => _$KorisnikToJson(this);
}
